crates/ruff_python_parser/src/parser/pattern.rs

```diff
@@ -172,18 +207,33 @@ impl<'src> Parser<'src> {
         lhs
     }
 
+    /// Parses a mapping pattern.
+    ///
+    /// # Panics
+    ///
+    /// If the parser isn't positioned at a `{` token.
+    ///
+    /// See: <https://docs.python.org/3/reference/compound_stmts.html#mapping-patterns>
     fn parse_match_pattern_mapping(&mut self) -> ast::PatternMatchMapping {
         let start = self.node_start();
+        self.bump(TokenKind::Lbrace);
+
+        // Return an empty `PatternMatchMapping` when finding a `}` right after the `{`
+        if self.eat(TokenKind::Rbrace) {
+            return ast::PatternMatchMapping {
+                range: self.node_range(start),
+                keys: vec![],
+                patterns: vec![],
+                rest: None,
+            };
+        }
+
```

Comment:
What's the motivation for early returning when we see a `{}`. Doesn't the `parse_comma_separated_list` handle this automatically for us because it exits the loop immediately when seeing the `}` (list terminator)?

Reply:
Not much, mainly that it's good to have. Plus I saw this pattern elsewhere so thought it made sense here.